plugins/scaffolder-backend/src/scaffolder/actions/builtin/filesystem/rename.test.ts

```diff
@@ -62,57 +64,47 @@ describe('fs:rename', () => {
     });
   });
 
-  it('should throw an error when files is not an array', async () => {
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: undefined } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: {} } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: '' } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
-
-    await expect(
-      action.handler({
-        ...mockContext,
-        input: { files: null } as any,
-      }),
-    ).rejects.toThrow(/files must be an Array/);
+  it('should throw ZodError when files is not an array', async () => {
+    const testCases = [
+      { input: undefined },
+      { input: {} },
+      { input: '' },
+      { input: null },
+    ];
+
+    await Promise.all(
+      testCases.map(({ input }) =>
+        expect(
+          Promise.resolve().then(() =>
+            action.handler({
+              ...mockContext,
+              input: { files: input } as any,
```

Comment:
There's no need to write unit tests for types that Zod will reject. Zod itself is the (runtime) test. In the code after Zod's `parse`, we know that any types that don't satisfy it would not go through. Anything that might go through wrongly, should be written as a unit test for that subsequent code.

This is the same logic as once we are in TypeScript, we no longer write a whole class of type-checking unit tests that were the norm for JavaScript. Not only does they not help prevent regression, they are more ugly to write in TypeScript than JavaScript.